src/controllers/user/signin.ts

```diff
@@ -0,0 +1,19 @@
+import type { Request, Response } from 'express';
+import User from '../../database/user';
+
+const signIn = async (req:Request, res:Response): Promise<void> => {
+  try {
+    const { username, password } = req.body;
+    const user = await User.findOne({ username });
+    if (user?.password === password) {

```

Comment:
password must be hashed in the db, you should hash the incoming password and compare hashes

Reply:
sign in need to be coded from stratch